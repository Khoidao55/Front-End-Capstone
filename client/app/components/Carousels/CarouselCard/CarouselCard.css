#card {
  position: relative;
  width: 300px;
  height: 575px;
  background-color: white;
  border: 2px solid black;
}

#product-image {
  border-bottom: 1px solid black;
}

.descriptive {
  padding-left: 10px;
}

#product-name {
  margin: 10px;
}

#price {
  color: gray;
}

#action {
  position: absolute;
  right: 5px;
  top: 5px;
  z-index: 2;
}
