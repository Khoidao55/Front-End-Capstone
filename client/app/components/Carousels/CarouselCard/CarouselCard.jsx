/* eslint-disable import/extensions */
/* eslint-disable jsx-a11y/control-has-associated-label */
import React from 'react';
import propTypes from 'proptypes';
import StarRating from '../../StarRating/StarRating.jsx';
import './CarouselCard.css';

const CarouselCard = ({
  id,
  productImage = 'https://watertownbusinesscoalition.com/assets/images/no_image_available.jpeg',
  category,
  name,
  originalPrice,
  salePrice,
  reviewScore,
  handleActionButton,
  handleRedirect,
  carouselType,
  features,
}) => {
  let displayPrice;

  if (salePrice !== null) {
    displayPrice = (
      <>
        <span className="descriptive old-price">{`$${originalPrice}`}</span>
        <span className="descriptive display-price">{`$${salePrice}`}</span>
      </>
    );
  } else {
    displayPrice = <span className="descriptive" id="display-price">{`$${originalPrice}`}</span>;
  }

  return (
    <div className={`card ${carouselType}-card`} onClick={() => { handleRedirect(id) }} >
      <button
        aria-label="Action Button"
        className="card-action-btn"
        type="button"
        onClick={(e) => { e.stopPropagation(); handleActionButton(id, name, features); }}
      >
        <i className={
          carouselType === 'related'
            ? 'related-action-btn fas fa-star'
            : 'outfit-action-btn far fa-times-circle'
        }
        />
      </button>
      <div className="card-image-container">
        {productImage
          ? <img className="card-image" src={productImage} alt={name ? name : 'Product Name'} />
          : (
            <img
              className="card-no-image"
              src="https://watertownbusinesscoalition.com/assets/images/no_image_available.jpeg"
              alt={name ? name : 'Product Name'}
            />
          )}
      </div>
      <div className="descriptive-container">
        <span className="descriptive category">{category ? category : 'Category'}</span>
        <h3 className="descriptive product-name">{name ? name : 'Product Name'}</h3>
        {displayPrice}
        <StarRating reviewScore={reviewScore} setMargin="10px 0 0 10px" />
      </div>
    </div>
  );
};

CarouselCard.propTypes = {
  id: propTypes.number.isRequired,
  productImage: propTypes.string,
  category: propTypes.string.isRequired,
  name: propTypes.string.isRequired,
  originalPrice: propTypes.string.isRequired,
  salePrice: propTypes.string,
  reviewScore: propTypes.number,
  handleActionButton: propTypes.func.isRequired,
  handleRedirect: propTypes.func.isRequired,
  carouselType: propTypes.string.isRequired,
};

export default CarouselCard;
