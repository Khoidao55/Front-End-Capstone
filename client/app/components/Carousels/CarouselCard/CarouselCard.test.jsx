import React from 'react';
import { shallow } from 'enzyme';
import CarouselCard from './CarouselCard.jsx';

const dummyCard = {
  id: 1234,
  productImage: 'https://bsaber.com/wp-content/uploads/2020/01/a2040f737cc603cf561c54624bde21715b7cbf52.jpg',
  category: 'meme',
  name: 'pretty cool bananas',
  originalPrice: '1234567',
  salePrice: '123',
  stars: '3.5',
  handleActionButton: () => { console.log('Action!'); },
  handleRedirect: () => { console.log('Redirect!'); },
  carouselType: 'related',
  features: [{ feature: 'bananas', value: 'pretty cool' }],
};

describe('CarouselCard', () => {
  test('CarouselCard component renders correctly', () => {
    const wrapper = shallow(<CarouselCard { ...dummyCard } />);

    expect(wrapper).toMatchSnapshot();
  });
});
