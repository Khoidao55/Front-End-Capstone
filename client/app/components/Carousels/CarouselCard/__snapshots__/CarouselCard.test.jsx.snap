// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselCard CarouselCard component renders correctly 1`] = `ShallowWrapper {}`;
